src/Api/Serializer/RestSerializer.php

```diff
@@ -127,6 +127,16 @@ private function applyHeader($name, Shape $member, $value, array &$opts)
             $value = TimestampShape::format($value, 'rfc822');
         }
 
+        if ($member['jsonvalue']) {
+            $value = json_encode(json_decode($value));
```

Comment:
This assumes that jsonvalue headers are being passed in as JSON-encoded strings. I don't think that's a correct assumption.